tpchgen/tests/integration_tests.rs

```diff
@@ -19,14 +19,9 @@ fn read_tbl_gz<P: AsRef<Path>>(path: P) -> Vec<String> {
         .expect("Failed to read lines")
 }
 
-fn test_generator<T, G, F>(
-    generator_fn: F,
-    reference_path: &str,
-    scale_factor: f64,
-    transform_fn: impl Fn(T) -> String,
-) where
-    G: Iterator<Item = T>,
-    F: FnOnce(f64) -> G,
+fn test_generator<T, I>(iter: I, reference_path: &str, transform_fn: impl Fn(T) -> String)
+where
+    I: Iterator<Item = T>,
```

Comment:
I remove the closure being passed in as all it was doing was creating an iterator using an sf (which was also being passed in). This just accepts an iterator now.